code/datums/elements/area_based_godmode.dm

```diff
@@ -0,0 +1,83 @@
+/**
+ * Area-based godmode.
+ * (area, allow_area_subtypes, gain_message, lose_message)
+ */
+/datum/element/area_based_godmode
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 1
+
+	/// The type of area that will trigger godmode.
+	var/area_type
+
+	/// Whether or not to allow subtypes of the area type to trigger godmode.
+	var/allow_area_subtypes
+
+	/// The message to send to the mob when they gain godmode.
+	var/gain_message
+
+	/// The message to send to the mob when they lose godmode.
+	var/lose_message
+
+	/// Lazy. The cache of mobs that are currently in godmode; prevents sending the gain message multiple times.
+	var/list/godmode_cache
```

Comment:
you sure this shouldn't be a component?

Reply:
I don't even see why we need to track everyone with godmode, surely we can just check if... they have godmode or not to determine whether they get the message

Reply:
> I don't even see why we need to track everyone with godmode, surely we can just check if... they have godmode or not to determine whether they get the message

💭